src/PerformanceSensitiveAnalyzers/UnitTests/AvoidAllocationWithArrayEmptyCodeFixTests.cs

```diff
@@ -0,0 +1,550 @@
+        [InlineData("IEnumerable<int>")]
+        [InlineData("IReadOnlyList<int>")]
+        [InlineData("IReadOnlyCollection<int>")]
+        public async Task ShouldReplaceEmptyListCreationWithArrayEmptyWithReturnType(string type)
+        {
+            var initial = $@"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{{
+    class SampleClass
+    {{
+        [PerformanceSensitive(""uri"")]
+        public {type} DoSomething()
+        {{
+            return new List<int>();
+        }}
+    }}
+}}";
+
+            var expected = $@"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{{
+    class SampleClass
+    {{
+        [PerformanceSensitive(""uri"")]
+        public {type} DoSomething()
+        {{
+            return Array.Empty<int>();
+        }}
+    }}
+}}";
+            await VerifyCS.VerifyCodeFixAsync(initial, VerifyCS.Diagnostic(ExplicitAllocationAnalyzer.ObjectCreationRule).WithSpan(13, 20, 13, 35), expected);
+        }
+
+        [Fact]
+        public async Task ShouldReplaceEmptyListCreationWithArrayEmptyWhenReturnFromMethodArray()
+        {
+            var before = @"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{
+    class SampleClass
+    {
+        [PerformanceSensitive(""uri"")]
+        public int[] DoSomething()
+        {
+            return new int[0];
+        }
+    }
+}";
+            var after = @"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{
+    class SampleClass
+    {
+        [PerformanceSensitive(""uri"")]
+        public int[] DoSomething()
+        {
+            return Array.Empty<int>();
+        }
+    }
+}";
+            
+            await VerifyCS.VerifyCodeFixAsync(before, VerifyCS.Diagnostic(ExplicitAllocationAnalyzer.ArrayCreationRule).WithSpan(13, 20, 13, 30), after);
+        }
+
+        [Fact]
+        public async Task should_replace_empty_list_creation_with_array_empty_for_arrow_expression()
+        {
+            var before = @"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{
+    class SampleClass
+    {
+        [PerformanceSensitive(""uri"")]
+        public IEnumerable<int> DoSomething => new List<int>();
+    }
+}";
+            var after = @"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{
+    class SampleClass
+    {
+        [PerformanceSensitive(""uri"")]
+        public IEnumerable<int> DoSomething => Array.Empty<int>();
+    }
+}";
+
+            await VerifyCS.VerifyCodeFixAsync(before, VerifyCS.Diagnostic(ExplicitAllocationAnalyzer.ObjectCreationRule).WithSpan(11, 48, 11, 63), after);
+        }
+        [Fact]
+        public async Task should_replace_empty_list_creation_with_array_empty_for_readonly_property()
```

Comment:
rename?

Reply:
Applies to all unit tests in this file

Reply:
We probably need a codefix to convert between Pascal and kebab-case.

Reply:
snake_case here